src/js/background/assignManager.js

```diff
@@ -244,15 +244,18 @@ const assignManager = {
   },
 
   async getPermissions() {
-    const {permissions} = await browser.permissions.getAll();
-    permissions.includes("bookmarks") ? this.makeBookmarksMenu() : browser.contextMenus.remove(this.OPEN_IN_CONTAINER);
-  },
-
-  makeBookmarksMenu() {
-    this.initBookmarksMenu();
-    browser.contextualIdentities.onCreated.addListener(this.contextualIdentityCreated);
-    browser.contextualIdentities.onUpdated.addListener(this.contextualIdentityUpdated);
-    browser.contextualIdentities.onRemoved.addListener(this.contextualIdentityRemoved);
+    const hasPermission = await browser.permissions.contains({permissions: ["bookmarks"]});
+    if (hasPermission) {
```

Comment:
To avoid the issue of attempting to add an already-added menu twice, remember whether you've already shown the menu, e.g. like this:

```javascript
if (this.hadBookmark === hasPermission) {
  return;
}
this.hadBookmark = hasPermission;
```

Reply:
@Rob--W I'm not sure I see how the menu could be added twice. `getPermissions()`, or `resetBookmarksMenuItem()` now, only runs once when the containers extension loads and then when the checkbox is toggled for permissions in the options menu. When the containers are added/removed/updated, it only changes the single item that changed in the menu. Do you have an example of when this could happen?

Reply:
This could have happened in your previous patch, where a `return` was missing.

This can still happen when a user opens `about:addons` plus the options page twice. The checkbox states are not updated when the permission is changed elsewhere, so when you select a checkbox in one options page, the event will fire in the background page. When you check the checkbox in the second options page, the event will fire once more.